sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImplTest.java

```diff
@@ -421,25 +406,22 @@ public void testInsertFailsGracefully() throws Exception {
         .setInsertErrors(ImmutableList.of(new InsertErrors().setIndex(0L)));
 
     when(response.getContentType()).thenReturn(Json.MEDIA_TYPE);
+    // Always return 200.
     when(response.getStatusCode()).thenReturn(200);
-    when(response.getContent()).thenAnswer(new Answer<InputStream>() {
-      boolean first = true;
-      @Override
-      public InputStream answer(InvocationOnMock invocation) throws Throwable {
-        if (first) {
-          first = false;
-          return toStream(row1Failed);
-        }
-        return toStream(row0Failed);
-      }
-    });;
+    // Return row 1 failing, then we retry row 1 as row 0, and row 0 persistently fails.
+    when(response.getContent())
+        .thenReturn(toStream(row1Failed))
+        .thenReturn(toStream(row0Failed))
```

Comment:
Whatever the last thenReturn is used will be used for all subsequent calls so no point in repeating .thenReturn(toStream(row0Failed)) 3 times.